### PR TITLE
crt0: Fix bss init routine

### PR DESCRIPTION
**Pull request by F0bes**
There was a severe issue where the bss segment was not properly being zeroed.
This caused major issues when using ps2link reset. IE: graph vram allocations not being reset, or `setenv` crashing after a soft reset. 

I originally thought these were two different issues, but newlib uses a static local variable to track if it has previously allocated https://github.com/ps2dev/newlib/blob/ee-v4.4.0/newlib/libc/stdlib/setenv_r.c#L56
Obviously, if you soft reset the system without marking this flag as zero, the next time it tries to use this no-longer-allocated memory, you would get a crash.

TLDR: This fixes globals not being set to zero, which in turn fixes probably that bug you've been chasing for weeks.

### CODE REVIEW

**Comment by uyjulian** [comment]
Now that's a fun issue.

**Comment by davidmalcolm** [comment]
FWIW I've filed [this RFE](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=115496) against GCC to perhaps add a compile-time warning about such mistakes.

**Comment by F0bes** [comment]
Good idea! As we've both witnessed, this can cause quite a headache :^)
